examples/2_Intermediate/vmec_adjoint.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python
+
+import os
+import numpy as np
+import matplotlib.pyplot as plt
+from simsopt.mhd.vmec import Vmec
+from simsopt.mhd.vmec_diagnostics import IotaTargetMetric
+from simsopt.objectives.least_squares import LeastSquaresProblem
+from simsopt.solve.serial import least_squares_serial_solve
+
+"""
+Here, we perform an optimization begining with a 3 field period rotating ellipse
+boundary to obtain iota = 0.381966. The derivatives are obtained with an adjoint
+method. This is based on a published result in
+Paul, Landreman, and Antonsen, Journal of Plasma Physics (2021). The number of
+modes in the optimization space is slowly increased from |m|,|n| <= 2 to 5. At
+the end, the initial and final profiles are plotted.
+"""
+
+# target profile of rotational transform, here just a constant:
+target_function = lambda s: 0.381966
+adjoint_epsilon = 1.e-1  # perturbation amplitude for adjoint solve
+
+filename = os.path.join(os.path.dirname(__file__), 'inputs', 'input.rotating_ellipse')
+vmec = Vmec(filename, ntheta=100, nphi=100)
+
+vmec.run()
+iotas_init = vmec.wout.iotas
+
+obj = IotaTargetMetric(vmec, target_function, adjoint_epsilon)
+prob = LeastSquaresProblem([(obj, 0, 1)])
+
+surf = vmec.boundary
+surf.all_fixed(True)
+# Slowly increase range of modes in optimization space
+for max_mode in range(3, 6):
+    print(max_mode)
+    surf.fixed_range(mmin=0, mmax=max_mode,
+                     nmin=-max_mode, nmax=max_mode, fixed=False)
+
+    least_squares_serial_solve(prob, grad=True, ftol=1e-12, gtol=1e-12, xtol=1e-12)
```

Comment:
I have modified the example so that it only runs 1 function eval at low resolution if in CI. I have added it to examples/run_vmec_examples. I am now also using scipy.minimize directly rather than the least_squares_serial_solve.